### ISSUE TITLE
cluster wide secret "<argocd-name>-default-cluster-config" being created causes "Out of Sync"

### ISSUE THREAD

**Issue by nladha09** [redhat-developer/gitops-operator#189]
**Issue / Bug Description**
- We're using the openshift-gitops operator (v.1.1.2) as a "Parent" instance in a multi-instance setup where the ArgoCD in the openshift-gitops namespace is the parent instance, and it deploys (among other things) other instances of ArgoCD to local and remote clusters. This works fine, but we have observed some issues and have some questions.

- Our ArgoCD apps for deploying other instances of ArgoCD go to "Out of Sync" because of the presence of a secret (which we do not have in our synced Git repo). The secret is called <argocd-name>-default-cluster-config, so for example my-argocd-default-cluster-config. I am able to prune this object to make the app status go to "Synced" but it is recreated afterwards.

- I wasn't able to find any docs that explain this secret and what causes it to be created so would appreciate any information. If it is not sensible to remove this secret, what workarounds are possible to prevent it from making our apps go out of sync?

**Expected behavior**
- "Out of Sync" requires pruning (not expected). What is this cluster wide secret and can its creation be prevented?

**Additional context**
- ACM installs GitOps operator on cluster; create ArgoCD resource on management cluster; ArgoCD reconciles App of Apps creates apps. This was not working in previous version (v1.1.2) either.

**Comment by jannfis** [comment]
Also, can you please copy&paste the complete `metadata` section of the secret please?

**Comment by nladha09** [comment]
@jannfis - Thank you for your response - I think this may be fixed now!

*Follow-up Question*: I noticed a [similar case](https://issues.redhat.com/browse/GITOPS-1025) was raised and the resolution states that this issue is fixed in Red Hat OpenShift GitOps 1.2. I've tried this out and I don't see the same issue now - although I don't see the labels mentioned in your original message on the secret, it is no longer considered "Out of Sync" by ArgoCD. 

---

Please find below the requested `metadata` section in case it's useful:

~~~
metadata:
  creationTimestamp: '2021-08-18T14:23:38Z'
  labels:
    app.kubernetes.io/managed-by: test-tenant
    app.kubernetes.io/name: test-tenant-default-cluster-config
    app.kubernetes.io/part-of: argocd
    argocd.argoproj.io/secret-type: cluster
    mo/parent-argo: management-test-tenant-tooling
  managedFields:
    - apiVersion: v1
      fieldsType: FieldsV1
      fieldsV1:
        'f:data':
          .: {}
          'f:config': {}
          'f:name': {}
          'f:namespaces': {}
          'f:server': {}
        'f:metadata':
          'f:labels':
            .: {}
            'f:app.kubernetes.io/managed-by': {}
            'f:app.kubernetes.io/name': {}
            'f:app.kubernetes.io/part-of': {}
            'f:argocd.argoproj.io/secret-type': {}
            'f:mo/parent-argo': {}
        'f:type': {}
      manager: gitops-operator
      operation: Update
      time: '2021-08-18T14:23:38Z'
~~~

**Comment by nladha09** [comment]
issue may be fixed; adding metadata as requested

**Comment by jannfis** [comment]
We commited a [fix](https://github.com/argoproj-labs/argocd-operator/commit/516e937f537b4463965c941290ca51477a03bb1e) at the upstream operator that will prevent these situations from occuring. Will be released with v1.3.

**Comment by nladha09** [comment]
This is great news - Thank you for update!
